lib/jetstream/pull_consumer.ex

```diff
@@ -71,13 +69,21 @@ defmodule Jetstream.PullConsumer do
   @callback handle_message(message :: Jetstream.message()) ::
               :ack | :nack | :noreply
 
-  @type settings :: %{
-          :connection_name => GenServer.server(),
-          :stream_name => binary(),
-          :consumer_name => binary(),
-          optional(:connection_retry_timeout) => pos_integer(),
-          optional(:connection_retries) => pos_integer()
-        }
+  @type settings ::
+          %{
+            :connection_name => GenServer.server(),
+            :stream_name => binary(),
+            :consumer_name => binary(),
+            optional(:connection_retry_timeout) => pos_integer(),
+            optional(:connection_retries) => pos_integer()
+          }
+          | [
+              connection_name: GenServer.server(),
+              stream_name: binary(),
+              consumer_name: binary(),
+              connection_retry_timeout: pos_integer(),
+              connection_retries: pos_integer()
+            ]
```

Comment:
Correct me if I'm wrong, but I think there's no way to emphasize that some Keyword list keys are required and others are optional.

I wonder which version is more correct - the current one or

```elixir
@type option ::
          {:connection_name, GenServer.server()}
          | {:stream_name, binary()}
          | {:consumer_name, binary()}
          | {:connection_retry_timeout, pos_integer()}
          | {:connection_retries, pos_integer()}

  @type settings ::
          %{
            :connection_name => GenServer.server(),
            :stream_name => binary(),
            :consumer_name => binary(),
            optional(:connection_retry_timeout) => pos_integer(),
            optional(:connection_retries) => pos_integer()
          }
          | [option()]
```

Reply:
Usually what I observed is that type specs are unspecified (I.e. `Keyword.t() | map()`), while parameters are only documented as prose in `@moduledoc`